msgpack-core/src/main/java/org/msgpack/core/MessagePacker.java

```diff
@@ -1129,6 +1129,35 @@ public MessagePacker writePayload(byte[] src, int off, int len)
         return this;
     }
 
+    /**
+     * This method copies the lower 8-bits of the characters.
+     * A separate writePayload(String src) method would be more performant because we're able to
+     * directly copy the bytes from the string to the right location in MessageBuffer.  Introducing
+     * encoding in an attempt to re-use the existing writePayload(byte[] src) creates unnecessary overhead
+     * which in turn will affect performance.
+     *
+     * @param src the string data to add
+     * @return this
+     * @throws IOException
+     */
+
+    public MessagePacker writePayload(String src) throws IOException
+    {
+        int len = src.length();
```

Comment:
If the input string contains non-ascii characters, src.getBytes() will produce a longer array than src.length(), so using this method can easily create corrupted data.